### PR TITLE
Fix json tests

### LINKED ISSUE #3837: Intermittent failure with NewtonsoftJson test
Intermittent failure with GraphQL.Tests.Execution.InputConversionNewtonsoftJsonTests.can_convert_json_to_input_object_with_custom_converter

https://github.com/graphql-dotnet/graphql-dotnet/actions/runs/7227876411/job/19696340154?pr=3833

### PR DESCRIPTION
**Pull request by Shane32**
Fixes #3837

### CODE REVIEW

File: src/GraphQL.Tests/Execution/InputConversionSystemTextJsonTests.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 namespace GraphQL.Tests.Execution;
 
+[Collection("StaticTests")]
```
Comment (Shane32) [inline src/GraphQL.Tests/Execution/InputConversionSystemTextJsonTests.cs:3]: The attribute was defined on the base class but I'm guessing it must not propagate to derived classes.

File: src/GraphQL.Tests/Execution/InputConversionNewtonsoftJsonTests.cs [hunk 2]
```diff
@@ -1,6 +1,9 @@
 namespace GraphQL.Tests.Execution;
 
+[Collection("StaticTests")]
 public class InputConversionNewtonsoftJsonTests : InputConversionTestsBase
 {
-    protected override Inputs VariablesToInputs(string variables) => variables.ToInputs();
+    private static readonly IGraphQLTextSerializer _serializer = new NewtonsoftJson.GraphQLSerializer(indent: true);
+    protected override Inputs VariablesToInputs(string variables)
+        => _serializer.Deserialize<Inputs>(variables) ?? Inputs.Empty;
```
Comment (gao-artur) [inline src/GraphQL.Tests/Execution/InputConversionNewtonsoftJsonTests.cs:8]: Maybe do the same in STJ tests for consistency and to make the use of required serializer explicit?

**Review by gao-artur — APPROVED** [review]

File: src/GraphQL.Tests/Execution/InputConversionNewtonsoftJsonTests.cs (hunk 2 above)
Comment (Shane32) [reply src/GraphQL.Tests/Execution/InputConversionNewtonsoftJsonTests.cs:8]: Good idea

Comment (Shane32) [reply src/GraphQL.Tests/Execution/InputConversionNewtonsoftJsonTests.cs:8]: Done

**Pull request merged**